tests/x11/evolution.pm

```diff
@@ -20,11 +20,18 @@ sub run {
     x11_start_program("evolution");
     my @tags = qw(test-evolution-1 evolution-default-client-ask);
     push(@tags, 'evolution-preview-release') if is_gnome_next;
-    do {
-        assert_screen \@tags;
-        send_key 'alt-o'                                  if match_has_tag('evolution-preview-release');
-        assert_and_click 'evolution-default-client-agree' if match_has_tag('evolution-default-client-ask');
-    } until (match_has_tag('test-evolution-1'));
+    my $times = scalar(@tags) - 1;
+    assert_screen \@tags;
+
+    unless (match_has_tag('test-evolution-1')) {
+        for (0 .. $times) {
+            assert_and_click 'evolution-default-client-agree' if match_has_tag('evolution-default-client-ask');
+            send_key 'alt-o'                                  if match_has_tag('evolution-preview-release');
```

Comment:
I wonder if we can have else if here and spare one check of tag, if click is done.